pop/service/api.go

```diff
@@ -0,0 +1,215 @@
+}
+
+// FinalStatement is the final configuration holding all data necessary
+// for a verifier.
+type FinalStatement struct {
+	// Desc is the description of the pop-party.
+	Desc *PopDesc
+	// Attendees holds a slice of all public keys of the attendees.
+	Attendees []abstract.Point
+	// Signature is created by all conodes responsible for that pop-party
+	Signature []byte
+}
+
+// The toml-structure for (un)marshaling with toml
+type finalStatementToml struct {
+	Desc      *popDescToml
+	Attendees []string
+	Signature string
+}
+
+// NewFinalStatementFromString creates a final statement from a string.
+func NewFinalStatementFromString(s string) *FinalStatement {
+	fsToml := &finalStatementToml{}
+	_, err := toml.Decode(s, fsToml)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	sis := []*network.ServerIdentity{}
+	for _, s := range fsToml.Desc.Roster {
+		uid, err := uuid.FromString(s[2])
+		if err != nil {
+			log.Error(err)
+			return nil
+		}
+		sis = append(sis, &network.ServerIdentity{
+			Address:     network.Address(s[0]),
+			Description: s[1],
+			ID:          network.ServerIdentityID(uid),
+			Public:      B64ToPoint(s[3]),
+		})
+	}
+	rostr := onet.NewRoster(sis)
+	desc := &PopDesc{
+		Name:     fsToml.Desc.Name,
+		DateTime: fsToml.Desc.DateTime,
+		Location: fsToml.Desc.Location,
+		Roster:   rostr,
+	}
+	atts := []abstract.Point{}
+	for _, p := range fsToml.Attendees {
+		atts = append(atts, B64ToPoint(p))
+	}
+	sig := make([]byte, 64)
+	sig, err = base64.StdEncoding.DecodeString(fsToml.Signature)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	return &FinalStatement{
+		Desc:      desc,
+		Attendees: atts,
+		Signature: sig,
+	}
+}
+
+// PointToB64 converts an abstract.Point to a base64-point.
+func PointToB64(p abstract.Point) string {
+	pub, err := p.MarshalBinary()
+	if err != nil {
+		log.Error(err)
+		return ""
+	}
+	return base64.StdEncoding.EncodeToString(pub)
+}
+
+// B64ToPoint converts a base64-string to an abstract.Point.
+func B64ToPoint(str string) abstract.Point {
+	public := network.Suite.Point()
+	buf, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	err = public.UnmarshalBinary(buf)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	return public
+}
+
+// ScalarToB64 converts an abstract.Scalar to a base64-string.
+func ScalarToB64(s abstract.Scalar) string {
+	sec, err := s.MarshalBinary()
+	if err != nil {
+		log.Error(err)
+		return ""
+	}
+	return base64.StdEncoding.EncodeToString(sec)
+}
+
+// B64ToScalar converts a base64-string to an abstract.Scalar.
+func B64ToScalar(str string) abstract.Scalar {
+	scalar := network.Suite.Scalar()
+	buf, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	err = scalar.UnmarshalBinary(buf)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	return scalar
+}
+
+// ToToml returns a toml-string.
+func (fs *FinalStatement) ToToml() string {
```

Comment:
Same for here, returning an error seems way safer.. or if reluctant, at least put a comment saying that the return value can be empty and that means an invalid statement or stg.